ipynb/android-wloads/Android_PCMark_Tests.ipynb

```diff
@@ -0,0 +1,440 @@
+      "2016-03-23 13:33:10,781 INFO    :        TestEnv - Experiment results available also in:\n",
+      "2016-03-23 13:33:10,782 INFO    :        TestEnv -    /home/micgio01/work/lisa/results_latest\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Initialize a test environment using:\n",
+    "# the provided target configuration (my_target_conf)\n",
+    "# the provided test configuration   (my_test_conf)\n",
+    "te = TestEnv(target_conf=my_target_conf, test_conf=my_tests_conf)\n",
+    "target = te.target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Support Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This set of support functions will help us running the benchmark using different CPUFreq governors."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def set_performance():\n",
+    "    target.cpufreq.set_all_governors('performance')\n",
+    "\n",
+    "def set_powersave():\n",
+    "    target.cpufreq.set_all_governors('powersave')\n",
+    "\n",
+    "def set_interactive():\n",
+    "    target.cpufreq.set_all_governors('interactive')\n",
+    "\n",
+    "def set_sched():\n",
+    "    target.cpufreq.set_all_governors('sched')\n",
+    "\n",
+    "def set_ondemand():\n",
+    "    target.cpufreq.set_all_governors('ondemand')\n",
+    "    \n",
+    "    for cpu in target.list_online_cpus():\n",
+    "        tunables = target.cpufreq.get_governor_tunables(cpu)\n",
+    "        target.cpufreq.set_governor_tunables(\n",
+    "            cpu,\n",
+    "            'ondemand',\n",
+    "            **{'sampling_rate' : tunables['sampling_rate_min']}\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# CPUFreq configurations to test\n",
+    "confs = {\n",
+    "#    'performance' : {\n",
+    " #       'label' : 'prf',\n",
+    "  #      'set' :  set_performance,\n",
+    "   # },\n",
+    "#    'powersave' : {\n",
+    " #       'label' : 'pws',\n",
+    "  #      'set' :  set_powersave,\n",
+    "   # },\n",
+    "     'interactive' : {\n",
+    "         'label' : 'int',\n",
+    "         'set' :  set_interactive,\n",
+    "     },\n",
+    "    'sched' : {\n",
+    "        'label' : 'sch',\n",
+    "        'set' :  set_sched,\n",
+    "    },\n",
+    "#    'ondemand' : {\n",
+    " #       'label' : 'odm',\n",
+    "  #      'set' :  set_ondemand,\n",
+    "   # }\n",
+    "}\n",
+    "\n",
+    "# The set of results for each comparison test\n",
+    "results = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def pcmark_run(exp_dir):\n",
```

Comment:
Before that cell I would suggest to add a simple check that the required benchmark is already installed on target.
Something like:

```python
def check_packages(pkgname):
    try:
        output = target.execute('pm list packages -f | grep -i {}'.format(pkgname))
    except Exception:
        raise RuntimeError('Package: [{}] not availabe on target'.format(pkgname))

# Check for specified PKG name being available on target
check_packages('com.futuremark.pcmark.android.benchmark')
```

This should be just enough to stop the notebook execution with a suitable error which tells the user why the benchmark cannot be executed.